src/main/java/frc/robot/subsystems/vision/Limelight.java

```diff
@@ -0,0 +1,92 @@
+package frc.robot.subsystems.vision;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.net.PortForwarder;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import frc.robot.subsystems.LoggedSubsystem;
+
+import java.util.Optional;
+import java.util.OptionalDouble;
+
+public class Limelight extends LoggedSubsystem<LimelightLogInputs> {
+    public static Limelight INSTANCE = null;
+
+    private final NetworkTable table = NetworkTableInstance.getDefault().getTable("limelight");
+    private final NetworkTableEntry tx = table.getEntry("tx");
+    private final NetworkTableEntry ty = table.getEntry("ty");
+    private final NetworkTableEntry tv = table.getEntry("tv");
+    private final NetworkTableEntry ts = table.getEntry("ts");
+    private final NetworkTableEntry tid = table.getEntry("tid");
```

Comment:
The NetworkTables protocol was upgraded to version 4 this year. [Here](https://docs.wpilib.org/en/stable/docs/software/networktables/nt4-migration-guide.html) is a guide on how to migrate to the new version.